File: sdk/runanywhere-react-native/src/Core/Models/STT/STTTranscriptionResult.ts
/**
 * STTTranscriptionResult.ts
 * Matches iOS: Features/STT/Models/STTTranscriptionResult.swift
 */

/**
 * Timestamp information for individual words in transcription
 */
export interface TimestampInfo {
  /** The word */
  word: string;
  /** Start time in seconds */
  startTime: number;
  /** End time in seconds */
  endTime: number;
  /** Confidence score for this word (optional) */
  confidence?: number;
}

/**
 * Alternative transcription with confidence score
 */
export interface AlternativeTranscription {
  /** The alternative transcript text */
  transcript: string;
  /** Confidence score for this alternative */
  confidence: number;
}

/**
 * Transcription result from STT service
 * Matches iOS STTTranscriptionResult struct
 */
export interface STTTranscriptionResult {
  /** The transcribed text */
  transcript: string;
  /** Confidence score for the transcription (optional) */
  confidence?: number;
  /** Word-level timestamp information (optional) */
  timestamps?: TimestampInfo[];
  /** Detected language (optional) */
  language?: string;
  /** Alternative transcriptions (optional) */
  alternatives?: AlternativeTranscription[];
}

File: sdk/runanywhere-swift/Sources/RunAnywhere/Features/STT/Models/STTTranscriptionResult.swift
//
//  STTTranscriptionResult.swift
//  RunAnywhere SDK
//
//  Transcription result from STT service
//

import Foundation

// MARK: - STT Transcription Result

/// Transcription result from service
public struct STTTranscriptionResult: Sendable {
    public let transcript: String
    public let confidence: Float?
    public let timestamps: [TimestampInfo]?
    public let language: String?
    public let alternatives: [AlternativeTranscription]?

    public init(
        transcript: String,
        confidence: Float? = nil,
        timestamps: [TimestampInfo]? = nil,
        language: String? = nil,
        alternatives: [AlternativeTranscription]? = nil
    ) {
        self.transcript = transcript
        self.confidence = confidence
        self.timestamps = timestamps
        self.language = language
        self.alternatives = alternatives
    }

    // MARK: - Nested Types

    public struct TimestampInfo: Sendable {
        public let word: String
        public let startTime: TimeInterval
        public let endTime: TimeInterval
        public let confidence: Float?

        public init(word: String, startTime: TimeInterval, endTime: TimeInterval, confidence: Float? = nil) {
            self.word = word
            self.startTime = startTime
            self.endTime = endTime
            self.confidence = confidence
        }
    }

    public struct AlternativeTranscription: Sendable {
        public let transcript: String
        public let confidence: Float

        public init(transcript: String, confidence: Float) {
            self.transcript = transcript
            self.confidence = confidence
        }
    }
}
